packages/cli/src/models/files/ProjectFile.ts

```diff
@@ -22,53 +24,72 @@ interface ConfigFileCompilerOptions {
   };
 }
 
-export default class ProjectFile {
-  public fileName: string;
-  public data: {
-    name: string;
-    version: string;
-    manifestVersion: string;
-    dependencies: { [name: string]: string };
-    contracts: { [alias: string]: string };
-    publish: boolean;
-    compiler: ConfigFileCompilerOptions;
-  };
+interface ProjectFileData {
+  name: string;
+  version: string;
+  manifestVersion: string;
+  dependencies: { [name: string]: string };
+  contracts: { [alias: string]: string };
+  publish: boolean;
+  compiler: ConfigFileCompilerOptions;
+}
 
-  public static getLinkedDependencies(fileName: string = 'zos.json'): string[] {
-    const file = fs.parseJsonIfExists(fileName);
-    if (file && file.dependencies) {
-      return Object.keys(file.dependencies).map(
-        depName => `${depName}@${file.dependencies[depName]}`,
-      );
-    } else return [];
-  }
+export const PROJECT_FILE_NAME = 'project.json';
+export const PROJECT_FILE_PATH = path.join(
+  OPEN_ZEPPELIN_FOLDER,
+  PROJECT_FILE_NAME,
+);
+export const LEGACY_PROJECT_FILE_NAME = 'zos.json';
 
-  public constructor(fileName: string = 'zos.json') {
-    this.fileName = fileName;
-    try {
-      this.data = fs.parseJsonIfExists(this.fileName) || {
-        manifestVersion: MANIFEST_VERSION,
-      };
-      // if we failed to read and parse zos.json
-    } catch (e) {
-      e.message = `Failed to parse '${path.resolve(
-        fileName,
-      )}' file. Please make sure that ${fileName} is a valid JSON file. Details: ${
-        e.message
-      }.`;
-      throw e;
+export default class ProjectFile {
+  public filePath: string;
+  public data: ProjectFileData;
+
+  public static getLinkedDependencies(filePath: string = null): string[] {
+    const project = new ProjectFile(filePath);
+    if (project && project.dependencies) {
+      return Object.keys(project.dependencies).map(
+        depName => `${depName}@${project.dependencies[depName]}`,
+      );
+    }
+    return [];
+  }
+
+  public constructor(filePath: string = null) {
+    const defaultData = {
+      manifestVersion: MANIFEST_VERSION,
+    } as any;
+    // if project file path is provided use only it
+    if (filePath) {
+      this.filePath = filePath;
+      this.data = this.tryToLoadProject(filePath);
+    } else {
+      // try to load legacy file first
+      // TODO: remove legacy project file support as a breaking change at 3.0
+      let data = this.tryToLoadProject(LEGACY_PROJECT_FILE_NAME);
```

Comment:
I'm thinking about whether we should prioritize the legacy or the new format. What would be a scenario where both files are present? For instance, I'd expect an EVM package to use both: zos.json for supporting old versions of zOS, and the new format for new versions. If so, it would make sense to try the new format first, right?